module/scheduler_components/worker.cpp

```diff
@@ -179,19 +198,18 @@ void Worker::ProcessRemoteReadResult(const internal::RemoteReadResult& read_resu
       << "Transaction " << txn_id << " does not exist for remote read result";
 
   auto txn = txn_states_[txn_id].txn_holder->GetTransaction();
-  auto& rpp = txn_states_[txn_id].remote_passive_partitions;
 
-  if (rpp.count(read_result.partition()) > 0) {
-    rpp.erase(read_result.partition());
-    // Apply remote reads to local txn
-    for (const auto& key_value : read_result.reads()) {
-      (*txn->mutable_read_set())[key_value.first] = key_value.second;
-    }
+  txn_states_[txn_id].remote_reads_waiting_on -= 1;
+  if (read_result.will_abort()) {
+    txn->set_status(TransactionStatus::ABORTED);
+  }
+  // Apply remote reads to local txn
+  for (const auto& key_value : read_result.reads()) {
+    (*txn->mutable_read_set())[key_value.first] = key_value.second;
   }
 
   // If all remote reads arrived, move on with executing this transaction.
-  // Otherwise, notify the scheduler that this worker is free
-  if (rpp.empty()) {
+  if (txn_states_[txn_id].remote_reads_waiting_on == 0) {
```

Comment:
Not in this PR but a possible optimization here is to abort the txn as soon as the first abort remote read arrives. That also requires removing the CHECK above and garbage collecting "early remote reads" of non-existent transactions. 
Maybe add a TODO here as a reminder.